lib/RouterMixin.js

```diff
@@ -203,7 +203,12 @@ var RouterMixin = {
 };
 
 function join(a, b) {
-  return (a + b).replace(/\/\//g, '/');
+  if ((a+b).search(/^http/i) > -1) {
```

Comment:
Please observe the existing styles within this module and add comments explaining what you are doing.

Could you explain why you have changed this method? If the Link will now navigate via window.location.href, is this code ever called?